### ISSUE TITLE
Project not determined on Windows when projects are not using the default path

### ISSUE THREAD

**Issue by kvetis** [cyrilletuzi/vscode-angular-schematics#75]
Vs code
```
Version: 1.40.1 (user setup)
Commit: 8795a9889db74563ddd43eb0a897a2384129a619
Date: 2019-11-13T16:49:35.976Z
Electron: 6.1.2
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Windows_NT x64 10.0.18362
-----
Extension version: 2.2.0
```

The project is not determined correctly on my windows machine.
I've taken time to determine where the problem lies: 
On https://github.com/cyrilletuzi/vscode-angular-schematics/blob/b2382cc8869487ba2c5695b26d65d62453145998/src/schematics/generate.ts#L104

the `contextPath` is in unix format, because it was read from the `path` parameter of the `Context`, therefore it has a leading slash (`"/c:/repos/project-nx/apps/client/src/app/cards-modal"`. The `workspacePath` was read as `fsPath` and is in Windows format without leading slashes (`"c:\repos\project-nx"`) . Therefore the `indexOf` returns `-1` and the `workspacePath.length` does not have the leading unix `/`. The resulting substring, `projectPath` has therefore two characters extra from the project folder (`"nx/apps/client/src/app/cards-modal"`).

I tried switching reading `workspacePath` as `path` instead of `fsPath` or the other way around, but the rest of the extension code seems to be dependent on the formats of those two.

**Comment by cyrilletuzi** [comment]
Thanks for your report and your debugging.

I'll plan a fix for later this week.

As a side note, the code for paths management is indeed quite messy for several reasons (it was my first steps in VS Code extension authoring and as I use a Mac, it's difficult to test every single change on Windows). But I planned a general refactoring and automated tests (help welcomed about the last).

**Comment by cyrilletuzi** [comment]
Question: do you now if the issue was introduced by a newer version of VS Code? It seems strange that such a general issue didn't raised before.

**Comment by kvetis** [comment]
I've had the issue since I started working on a project this July and I always thought the problem was caused by a misconfiguration of the project rather by the extension. I'm not sure when it was the last time I had used the extension before that. Only today I became too bothered by the need to add --project everywhere, so I dug deeper.

**Comment by cyrilletuzi** [comment]
Ok, thanks.

**Comment by cyrilletuzi** [comment]
@kvetis Didn't forget you but my agenda was busier then expected.

While your debug seems right, I can't reproduce your issue. I'd need to have details of the directories structure, and which directory exactly is opened as the project inside VS Code.

**Comment by kvetis** [comment]
@cyrilletuzi I believe any workspace that does not have default `projects` as a project root demonstrates this issue. The project won't match https://github.com/cyrilletuzi/vscode-angular-schematics/blob/a58e6ab7563fd67c8ad13227657a526692e6ed93/src/schematics/generate.ts#L120.

I have created a [sample workspace](https://github.com/kvetis/vs-schematics-sample) using these steps:
```
ng new my-workspace --createApplication="false"
cd my-workspace
ng generate application app-one
ng generate application app-two
// rename `projects` to `apps`
// globaly find and replace `projects` with `apps`
// optionaly delete "defaultProject" in "angular.json"
```
You can just clone [the repo](https://github.com/kvetis/vs-schematics-sample) or follow the steps.

The angular workspace root is opened in VSCode.

**Comment by cyrilletuzi** [comment]
Thanks, reproduced. I'll plan a fix soon.

**Comment by cyrilletuzi** [comment]
This is fixed in 2.2.1, just released (may take a few minutes to appear). Can you confirm it fixes the issue in your case too?

Thanks a lot for your report and debug, that allowed me to do the fix quickly.

**Comment by kvetis** [comment]
@cyrilletuzi Great, I tested in our project and it works! Thx for the fix.
